accepted/00000-maintenance-windows.md

```diff
@@ -0,0 +1,390 @@
+ 503 | states.apply                                    | Apply states                                                                                       | Y
+ 504 | image.build                                     | Build an Image Profile                                                                             | N
+ 505 | image.inspect                                   | Inspect an Image                                                                                   | N
+ 506 | channels.subscribe                              | Subscribe to channels                                                                              | Y ?
+
+(That list is currently the full action list. Not all actions are still in use)
+
+
+### Schedule an Action
+
+When a system is assigned to a maintenance schedule, actions which are flagged as `maintenance_mode_only` should be executed only within a maintenance window.
+In the User Interface we should show in these cases not the datepicker, but a simple Combo box with the Date and Time of the next 20 Maintenance Windows.
+All actions will be scheduled at the begin of the Maintenance Window.
+
+When using SSM it is possible that systems have different maintenance windows. The combination systems belong
+to one maintenance schedule and systems without maintenance schedule is not critical. The action would be scheduled
+at a time which is inside of a maintenance window of the systems which belong to the schedule.
+
+Systems in different schedules needs special handling. We could implement 2 options:
+
+1. divide systems by maintenance schedules and display a combo box with the names. Let the user choose for which set he want to schedule it. He need to repeat it for the rest.
+2. As an alternative we provide via a Radio Button the Option "Next available Maintenance Window". This would apply for all systems in SSM.
+
+Scheduling an Action via XMLRPC outside of a maintenance window will result into an error.
+XMLRPC API called to list windows or get the next window can help to make the API usable.
+
+
+### Assign a Maintenance Schedule to a System
+
+There could be multiple ways to assign systems to maintenance schedules.
+
+- In the systems details page we need to show the selected schedule with a view and a change option.
+- It should be possible to change the maintenance schedule for multiple systems using SSM
+- In the details of the Maintenance Schedule we could have a systems page where add/remove is possible.
+
+If a Maintenance Schedule is assigned and which one, should be visible in the systems details page.
+It should be also part of the `system.getDetails()` XMLRPC API function.
+
+## System in Maintenance Mode
+
+The current status of the system should be reflected in the User Interface and API.
+
+If a system is in Maintenance Mode, the User Interface could show a special icon [ || ] (pause symbol of an audio player).
```

Comment:
Not sure if such an icon is a good idea, it could be ambiguous with virtualization pausing which some hypervisors support. I agree with the general idea to have it reflected in the UI somehow, just maybe we will want a different icon or just a label.

Reply:
Noted. Will discuss in the afternoon.